### PR TITLE
Added message button on profile page.

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
messageButton

### CODE REVIEW

File: src/client/src/Components/MessageButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import MailIcon from '@material-ui/icons/Mail';
+import { Link } from 'react-router-dom';
+
+interface MessagingProps {
```
Comment (johan1505) [inline src/client/src/Components/MessageButton/index.tsx]: Rename to MessageButtonProps

File: src/client/src/Components/MessageButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    email: string;
+}
+
+function MessageButton({ email }: MessagingProps) {
+    return (
+        <>
```
Comment (johan1505) [inline src/client/src/Components/MessageButton/index.tsx]: I think you can remove this empty html tag and just return the IconButton by itself

File: src/client/src/Domains/Messages/MessageRoot/index.tsx [hunk 3]
```diff
@@ -1,30 +1,58 @@
...
+import Loader from 'Components/Loader';
 import { IUser } from 'Domains/Accounts/api';
-
 import MessageSendForm from 'Domains/Messages/MessageSendForm';
 import Messages from 'Domains/Messages/GetMessages';
 import ConversationList from 'Domains/Messages/ConversationList';
+import { createMessage } from 'Domains/Messages/api';
+import { formInitialValues, INewMessageForm } from 'Domains/Messages/NewMessageForm';
+
+interface MessagingProps {
```
Comment (johan1505) [inline src/client/src/Domains/Messages/MessageRoot/index.tsx]: rename to MessageRootProps

File: src/client/src/Domains/Messages/MessageRoot/index.tsx [hunk 4]
```diff
@@ -1,30 +1,58 @@
...
+    const [sendRequest, isLoading] = useApi(request, {
+        onSuccess: (response) => {
+            setReceiver(response.data.user);
+        }
+    });
+
+    React.useEffect(() => {
+        // Check if email is populated with a valid email.
+        if (email && email.length > 0 && email !== ':email') {
+            setnewMessageForm({ email: email });
```
Comment (johan1505) [inline src/client/src/Domains/Messages/MessageRoot/index.tsx]: Change to ` setnewMessageForm({ email });`

**Review by johan1505 — CHANGES_REQUESTED** [review]

File: src/client/src/Components/MessageButton/index.tsx (hunk 1 above)
Comment (Mari-Hayashi) [reply src/client/src/Components/MessageButton/index.tsx]: Done.

File: src/client/src/Components/MessageButton/index.tsx (hunk 2 above)
Comment (Mari-Hayashi) [reply src/client/src/Components/MessageButton/index.tsx]: Done.

File: src/client/src/Domains/Messages/MessageRoot/index.tsx (hunk 3 above)
Comment (Mari-Hayashi) [reply src/client/src/Domains/Messages/MessageRoot/index.tsx]: Done.

File: src/client/src/Domains/Messages/MessageRoot/index.tsx (hunk 4 above)
Comment (Mari-Hayashi) [reply src/client/src/Domains/Messages/MessageRoot/index.tsx]: Done.

**Review by johan1505 — APPROVED** [review]

**Pull request merged**